### ISSUE TITLE
Pkg.test failed under Julia v0.7.0: MethodError: no method matching installed(::String)

### ISSUE THREAD

**Issue by BoundaryValueProblems** [JuliaIO/HDF5.jl#495]
I just installed Julia v0.7.0 and added HDF5.jl, which was successfully precompiled via ```using HDF5```. I could also generate some HDF5 files. However, the package test ```Pkg.test("HDF5")``` failed with the following error:
```
...
ERROR: LoadError: MethodError: no method matching installed(::String)
Closest candidates are:
  installed() at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:261
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] include(::String) at ./client.jl:398
 [6] top-level scope at none:0
in expression starting at /Users/xxx/.julia/packages/HDF5/PegIy/test/runtests.jl:13
ERROR: Package HDF5 errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/Types.jl:120
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] #test#61(::Bool, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/Operations.jl:1283
 [4] #test at ./none:0 [inlined]
 [5] #test#44(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:256
 [6] test at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:242 [inlined]
 [7] #test#43 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:239 [inlined]
 [8] test at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:239 [inlined]
 [9] #test#42 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:238 [inlined]
 [10] test at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:238 [inlined]
 [11] #test#41 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:237 [inlined]
 [12] test(::String) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v0.7/Pkg/src/API.jl:237
 [13] top-level scope at none:0
```
Thanks for your help!

**Comment by BoundaryValueProblems** [comment]
Today, I decided to rebuild the whole packages I installed so far via pkg> build (on Julia v0.7). Then, as before, when HDF5.jl was being built, it first tried to install gcc 8.2.0 via Homebrew.jl as before. It again took a long time (~ 1 hour or so), but somehow it finished today and built both gcc 8.2.0 via Homebrew.jl, and so did HDF5.jl. I have no idea why it worked this time, but not previously though. Previously, compiling gcc 8.2.0 failed. I’m using HDF5 v0.9.3 and Homebrew v0.7.0 (both the official releases, not the current master branches). I’m happy, but puzzled why it didn’t work before. On the other hand, I still cannot build Homebrew.jl as I reported at https://github.com/JuliaPackaging/Homebrew.jl/issues/245...

**Comment by musm** [comment]
Should work now ?

**Comment by BoundaryValueProblems** [comment]
@musm , yes, with HDF5.jl v0.10.0 under Julia v0.7.0, Pkg.test passed without any error. So, I'll close this. However, Homebrew.jl under v0.7.0 does not work as under v0.6.x; I cannot update that package at this point under Julia v0.7.0.

**Comment by musm** [comment]
Ok I will release  backports to the 0.9 release which will hopefully fix the 0.6 issues. Try again by tomorrow and let me know if you can finally upgrade.
